File: models/OrderItemDTO.js
const utils = require('../utils/utils');
const OrderItemDetailDTO = require('../models/OrderItemDetailDTO');
const OrderItemInstanceDTO = require('../models/OrderItemInstanceDTO');
const OrderItemInstanceType = require('../models/OrderItemInstanceType');
const OrderItemPromoDTO = require('../models/OrderItemPromoDTO');
const OrderItemSubsidyDTO = require('../models/OrderItemSubsidyDTO');
const OrderVatType = require('../models/OrderVatType');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}id`,
                label: `Идентификатор товара в заказе.  Позволяет идентифицировать товар в рамках данного заказа.  - [${labelPrefix}id]`,
                type: 'number',
            },
            {
                key: `${keyPrefix}offerId`,
                label: `Ваш SKU — идентификатор товара в вашей системе.  Разрешена любая последовательность длиной до 255 знаков.  Правила использования SKU:  * У каждого товара SKU должен быть свой.  * SKU товара нельзя менять — можно только удалить товар и добавить заново с новым SKU.  * Уже заданный SKU нельзя освободить и использовать заново для другого товара. Каждый товар должен получать новый идентификатор, до того никогда не использовавшийся в вашем каталоге.  [Что такое SKU и как его назначать](https://yandex.ru/support/marketplace/assortment/add/index.html#fields)  - [${labelPrefix}offerId]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}offerName`,
                label: `Название товара. - [${labelPrefix}offerName]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}price`,
                label: `Цена на товар в валюте заказа без учета вознаграждения партнеру за скидки по промокодам, купонам и акциям (параметр `subsidies`).  - [${labelPrefix}price]`,
                type: 'number',
            },
            {
                key: `${keyPrefix}buyerPrice`,
                label: `Цена на товар в валюте покупателя. В цене уже учтены скидки по:  * акциям; * купонам; * промокодам.  - [${labelPrefix}buyerPrice]`,
                type: 'number',
            },
            {
                key: `${keyPrefix}buyerPriceBeforeDiscount`,
                label: `Стоимость товара в валюте покупателя до применения скидок по:  * акциям; * купонам; * промокодам.  - [${labelPrefix}buyerPriceBeforeDiscount]`,
                type: 'number',
            },
            {
                key: `${keyPrefix}priceBeforeDiscount`,
                label: `{% note warning \"\" %}  Этот параметр устарел.  {% endnote %}  Стоимость товара в валюте магазина до применения скидок.  - [${labelPrefix}priceBeforeDiscount]`,
                type: 'number',
            },
            {
                key: `${keyPrefix}count`,
                label: `Количество единиц товара. - [${labelPrefix}count]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}vat`,
                ...OrderVatType.fields(`${keyPrefix}vat`, isInput),
            },
            {
                key: `${keyPrefix}shopSku`,
                label: `Ваш SKU — идентификатор товара в вашей системе.  Разрешена любая последовательность длиной до 255 знаков.  Правила использования SKU:  * У каждого товара SKU должен быть свой.  * SKU товара нельзя менять — можно только удалить товар и добавить заново с новым SKU.  * Уже заданный SKU нельзя освободить и использовать заново для другого товара. Каждый товар должен получать новый идентификатор, до того никогда не использовавшийся в вашем каталоге.  [Что такое SKU и как его назначать](https://yandex.ru/support/marketplace/assortment/add/index.html#fields)  - [${labelPrefix}shopSku]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}subsidy`,
                label: `{% note warning \"\" %}  Этот параметр устарел. Вместо него используйте `subsidies`.  {% endnote %}  Общее вознаграждение партнеру за DBS-доставку и все скидки на товар:  * по промокодам; * по купонам; * по баллам Плюса; * по акциям.  Передается в валюте заказа.  - [${labelPrefix}subsidy]`,
                type: 'number',
            },
            {
                key: `${keyPrefix}partnerWarehouseId`,
                label: `{% note warning \"\" %}  Этот параметр устарел. Не используйте его.  {% endnote %}  Идентификатор склада в системе партнера, на который сформирован заказ.  - [${labelPrefix}partnerWarehouseId]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}promos`,
                label: `[${labelPrefix}promos]`,
                children: OrderItemPromoDTO.fields(`${keyPrefix}promos${!isInput ? '[]' : ''}`, isInput, true), 
            },
            {
                key: `${keyPrefix}instances`,
                label: `[${labelPrefix}instances]`,
                children: OrderItemInstanceDTO.fields(`${keyPrefix}instances${!isInput ? '[]' : ''}`, isInput, true), 
            },
            {
                key: `${keyPrefix}details`,
                label: `[${labelPrefix}details]`,
                children: OrderItemDetailDTO.fields(`${keyPrefix}details${!isInput ? '[]' : ''}`, isInput, true), 
            },
            {
                key: `${keyPrefix}subsidies`,
                label: `[${labelPrefix}subsidies]`,
                children: OrderItemSubsidyDTO.fields(`${keyPrefix}subsidies${!isInput ? '[]' : ''}`, isInput, true), 
            },
            {
                key: `${keyPrefix}requiredInstanceTypes`,
                list: true,
                type: 'string',
                ...OrderItemInstanceType.fields(`${keyPrefix}requiredInstanceTypes`, isInput),
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'id': bundle.inputData?.[`${keyPrefix}id`],
            'offerId': bundle.inputData?.[`${keyPrefix}offerId`],
            'offerName': bundle.inputData?.[`${keyPrefix}offerName`],
            'price': bundle.inputData?.[`${keyPrefix}price`],
            'buyerPrice': bundle.inputData?.[`${keyPrefix}buyerPrice`],
            'buyerPriceBeforeDiscount': bundle.inputData?.[`${keyPrefix}buyerPriceBeforeDiscount`],
            'priceBeforeDiscount': bundle.inputData?.[`${keyPrefix}priceBeforeDiscount`],
            'count': bundle.inputData?.[`${keyPrefix}count`],
            'vat': bundle.inputData?.[`${keyPrefix}vat`],
            'shopSku': bundle.inputData?.[`${keyPrefix}shopSku`],
            'subsidy': bundle.inputData?.[`${keyPrefix}subsidy`],
            'partnerWarehouseId': bundle.inputData?.[`${keyPrefix}partnerWarehouseId`],
            'promos': utils.childMapping(bundle.inputData?.[`${keyPrefix}promos`], `${keyPrefix}promos`, OrderItemPromoDTO),
            'instances': utils.childMapping(bundle.inputData?.[`${keyPrefix}instances`], `${keyPrefix}instances`, OrderItemInstanceDTO),
            'details': utils.childMapping(bundle.inputData?.[`${keyPrefix}details`], `${keyPrefix}details`, OrderItemDetailDTO),
            'subsidies': utils.childMapping(bundle.inputData?.[`${keyPrefix}subsidies`], `${keyPrefix}subsidies`, OrderItemSubsidyDTO),
            'requiredInstanceTypes': utils.childMapping(bundle.inputData?.[`${keyPrefix}requiredInstanceTypes`], `${keyPrefix}requiredInstanceTypes`, OrderItemInstanceType),
        }
    },
}

File: src/gatling/scala/org/openapitools/client/model/OrderItemDTO.scala

package org.openapitools.client.model


case class OrderItemDTO (
    /* Идентификатор товара в заказе.  Позволяет идентифицировать товар в рамках данного заказа.  */
    _id: Option[Long],
    /* Ваш SKU — идентификатор товара в вашей системе.  Разрешена любая последовательность длиной до 255 знаков.  Правила использования SKU:  * У каждого товара SKU должен быть свой.  * SKU товара нельзя менять — можно только удалить товар и добавить заново с новым SKU.  * Уже заданный SKU нельзя освободить и использовать заново для другого товара. Каждый товар должен получать новый идентификатор, до того никогда не использовавшийся в вашем каталоге.  [Что такое SKU и как его назначать](https://yandex.ru/support/marketplace/assortment/add/index.html#fields)  */
    _offerId: Option[String],
    /* Название товара. */
    _offerName: Option[String],
    /* Цена на товар в валюте заказа без учета вознаграждения партнеру за скидки по промокодам, купонам и акциям (параметр `subsidies`).  */
    _price: Option[Number],
    /* Цена на товар в валюте покупателя. В цене уже учтены скидки по:  * акциям; * купонам; * промокодам.  */
    _buyerPrice: Option[Number],
    /* Стоимость товара в валюте покупателя до применения скидок по:  * акциям; * купонам; * промокодам.  */
    _buyerPriceBeforeDiscount: Option[Number],
    /* {% note warning \"\" %}  Этот параметр устарел.  {% endnote %}  Стоимость товара в валюте магазина до применения скидок.  */
    _priceBeforeDiscount: Option[Number],
    /* Количество единиц товара. */
    _count: Option[Integer],
    _vat: Option[OrderVatType],
    /* Ваш SKU — идентификатор товара в вашей системе.  Разрешена любая последовательность длиной до 255 знаков.  Правила использования SKU:  * У каждого товара SKU должен быть свой.  * SKU товара нельзя менять — можно только удалить товар и добавить заново с новым SKU.  * Уже заданный SKU нельзя освободить и использовать заново для другого товара. Каждый товар должен получать новый идентификатор, до того никогда не использовавшийся в вашем каталоге.  [Что такое SKU и как его назначать](https://yandex.ru/support/marketplace/assortment/add/index.html#fields)  */
    _shopSku: Option[String],
    /* {% note warning \"\" %}  Этот параметр устарел. Вместо него используйте `subsidies`.  {% endnote %}  Общее вознаграждение партнеру за DBS-доставку и все скидки на товар:  * по промокодам; * по купонам; * по баллам Плюса; * по акциям.  Передается в валюте заказа.  */
    _subsidy: Option[Number],
    /* {% note warning \"\" %}  Этот параметр устарел. Не используйте его.  {% endnote %}  Идентификатор склада в системе партнера, на который сформирован заказ.  */
    _partnerWarehouseId: Option[String],
    /* Информация о вознаграждениях партнеру за скидки на товар по промокодам, купонам и акциям. */
    _promos: Option[List[OrderItemPromoDTO]],
    /* Информация о маркировке единиц товара.  Возвращаются данные для маркировки, переданные в запросе [PUT campaigns/{campaignId}/orders/{orderId}/cis](../../reference/orders/provideOrderItemCis.md).  Если магазин еще не передавал коды для этого заказа, `instances` отсутствует.  */
    _instances: Option[List[OrderItemInstanceDTO]],
    /* Информация об удалении товара из заказа.  */
    _details: Option[List[OrderItemDetailDTO]],
    /* Список субсидий по типам. */
    _subsidies: Option[List[OrderItemSubsidyDTO]],
    /* Список необходимых маркировок товара. */
    _requiredInstanceTypes: Option[List[OrderItemInstanceType]]
)
object OrderItemDTO {
    def toStringBody(var_id: Object, var_offerId: Object, var_offerName: Object, var_price: Object, var_buyerPrice: Object, var_buyerPriceBeforeDiscount: Object, var_priceBeforeDiscount: Object, var_count: Object, var_vat: Object, var_shopSku: Object, var_subsidy: Object, var_partnerWarehouseId: Object, var_promos: Object, var_instances: Object, var_details: Object, var_subsidies: Object, var_requiredInstanceTypes: Object) =
        s"""
        | {
        | "id":$var_id,"offerId":$var_offerId,"offerName":$var_offerName,"price":$var_price,"buyerPrice":$var_buyerPrice,"buyerPriceBeforeDiscount":$var_buyerPriceBeforeDiscount,"priceBeforeDiscount":$var_priceBeforeDiscount,"count":$var_count,"vat":$var_vat,"shopSku":$var_shopSku,"subsidy":$var_subsidy,"partnerWarehouseId":$var_partnerWarehouseId,"promos":$var_promos,"instances":$var_instances,"details":$var_details,"subsidies":$var_subsidies,"requiredInstanceTypes":$var_requiredInstanceTypes
        | }
        """.stripMargin
}
